src/v2-router.js

```diff
@@ -0,0 +1,228 @@
+import express from 'express';
+import utils from 'naturescot-utils';
+import Registration from './controllers/v2/registration.js';
+
+const v2Router = express.Router();
+
+/**
+ * Clean an incoming PATCH request body to make it more compatible with the
+ * database and its validation rules.
+ *
+ * @param {any} body the incoming request's body
+ * @returns {any} cleanedBody a json object that's just got our cleaned up fields on it
+ */
+const cleanPatchInput = (body) => {
+  const cleanedBody = {};
+
+  // Check for the existence of each field and if found clean it if required and add to the cleanedBody object.
+  if (body.convictions) {
+    cleanedBody.convictions = body.convictions;
+  }
+
+  if (body.usingGL01) {
+    cleanedBody.usingGL01 = body.usingGL01;
+  }
+
+  if (body.usingGL02) {
+    cleanedBody.usingGL02 = body.usingGL02;
+  }
+
+  if (body.meatBaits) {
+    cleanedBody.meatBaits = body.meatBaits;
+  }
+
+  if (body.fullName) {
+    cleanedBody.fullName = body.fullName.trim();
+  }
+
+  if (body.addressLine1) {
+    cleanedBody.addressLine1 = body.addressLine1.trim();
+  }
+
+  if (body.addressLine2) {
+    cleanedBody.addressLine2 = body.addressLine2.trim();
+  }
+
+  if (body.addressTown) {
+    cleanedBody.addressTown = body.addressTown.trim();
+  }
+
+  if (body.addressCounty) {
+    cleanedBody.addressCounty = body.addressCounty.trim();
+  }
+
+  if (body.addressPostcode) {
+    cleanedBody.addressPostcode = body.addressPostcode.trim();
+  }
+
+  if (body.phoneNumber) {
+    cleanedBody.phoneNumber = body.phoneNumber.trim();
+  }
+
+  if (body.emailAddress) {
+    cleanedBody.emailAddress = utils.formatters.stripAndRemoveObscureWhitespace(body.emailAddress.toLowerCase());
```

Comment:
Likewise this should be validated too. The `validateAndFormatEmailAddress` function will throw an error, so the already mentioned 'catching errors in the `patch(/* ... */)` method' pattern will work for this too.

```js
if (body.emailAddress) {
  cleanedBody.emailAddress = utils.formatters.validateAndFormatEmailAddress(body.emailAddress);
}
```

Reply:
Resolved.